zip-0311.rst

```diff
@@ -0,0 +1,111 @@
+
+Light client
+  A client that is not a full participant in the network of Zcash peers. It can send and receive payments, but does not store or validate a copy of the blockchain.
+
+High probability
+  An event occurs with high probability if it occurs with probability 1-O(1/2^λ), where λ is the security parameter.
+
+Negligible probability
+  An event occurs with negligible probability if it occurs with probability O(1/2^λ), where λ is the security parameter.
+
+Merkle mountain range (MMR)
+  A Merkle mountain range (MMR) is binary hash tree that allows for efficient appends of new leaves without changing the value of existing nodes [#PeterTodd]_.
+
+Hard fork 
+  A hard fork is a change to a protocol that makes transactions from older clients invalid. 
+  
+
+Abstract
+========
+A proposed implementation of the probabilistic verification FlyClient protocol [#FlyClient]_ on top of the existing ZCash reference light client protocol [#ZIPXXX]_. This can be implemented with a hard fork to include a MMR proof in each block header.
+
+Motivation
+==========
+FlyClient reduces the number of block headers needed for light client verification of a valid chain, from linear (as in the current reference protocol) to logarithmic in blockchain length. This verification is correct with high probability. It also allows creation of subchain proofs, so light clients need only check blocks later than the most recently verified block index. Following that, verification of a transaction inclusion within that block follows the usual reference protocol [#ZIPXXX]_. 
+
+A smaller proof size could facilitate ‘side-chaining’ with other blockchains such as Ethereum. It also reduces bandwidth and storage requirements for limited clients like mobile or IoT devices.
+
+
+Specification
+=============
+
+Probabilistic sampling
+```````````````````````
+Instead of downloading every block header, the FlyClient samples blocks from the full node with the given sampling distribution. Here, x is the relative aggregate weight at the given block (e.g. if x=1/2 at some block, it means that half the chain's difficulty has been amassed in the interval up to that block), and δ is the fraction of the blocks which are always checked by the client with probability 1.
+
+.. figure:: pdf.png
+    :align: center
+    :scale: 50 %
+    :figclass: align-center
+    
+It has been found that as long as log_c(δ) is constant, where c is the adversary's mining power as a fraction of the honest mining power, then the number of light client queries needed are logarithmic in the size of the blockchain [#FlyClient]_.
+
+
+New block header
+`````````````````
+Each FlyClient prover has to maintain a commitment to all its block headers in a Merkle mountain range (MMR). This means that each ZCash block header must include the MMR root of all the blocks before it. This can be implemented in a hard fork of the ZCash protocol, in which the MMR root is added to the header of all old and new blocks. 
```

Comment:
This would need to specify the format of the MMR hash inputs and the location of the root field in the block header. However, this is enough for a draft.